.travis.yml

```diff
@@ -24,17 +25,19 @@ before_script:
   - export PATH="$PATH:$HOME/.cache/bin:$HOME/.cargo/bin"
   - GRPC_HEADER="$HOME/.cache/include/grpc/grpc.h"
   - if [[ $TRAVIS_OS_NAME == "osx" ]] && [[ ! -f $GRPC_HEADER ]]; then export CC=clang; brew update && brew install autoconf libtool shtool; fi
-  - test -f "$GRPC_HEADER" || (git clone -b v1.7.2 https://github.com/grpc/grpc && cd grpc && git submodule update --init && env prefix=$HOME/.cache make install_c)
+  - test -f "$GRPC_HEADER" || (git clone -b v1.10.0 https://github.com/grpc/grpc && cd grpc && git submodule update --init && env prefix=$HOME/.cache make install_c install_cxx)
   - export C_INCLUDE_PATH="$HOME/.cache/include"
+  - export CPLUS_INCLUDE_PATH="$HOME/.cache/include"
   - export LD_LIBRARY_PATH="$HOME/.cache/lib"
   - export DYLD_LIBRARY_PATH="$HOME/.cache/lib"
   - export LIBRARY_PATH="$HOME/.cache/lib"
   - export PKG_CONFIG_PATH="$HOME/.cache/lib/pkgconfig"
 script:
   - if [[ $TRAVIS_RUST_VERSION == "nightly" ]]; then cargo install -f clippy && cargo clippy; fi
   - export RUSTFLAGS=-Dwarnings
+  - export RUST_BACKTRACE=1
   - cargo build --no-default-features
   - cargo build --no-default-features --features protobuf-codec
   - cargo build
   - cargo test --all
-  - GRPCIO_SYS_USE_PKG_CONFIG=1 cargo test --all
+  - GRPCIO_SYS_USE_PKG_CONFIG=1 cargo build --all
```

Comment:
I have trouble in linking grpcio's test targets, need help. :sos:

Reply:
Can you give some more information?

Reply:
From: https://travis-ci.org/overvenus/grpc-rs/jobs/400903660
```
   Compiling grpcio-sys v0.2.3 (file:///home/travis/build/overvenus/grpc-rs/grpc-sys)
   Compiling grpcio v0.3.0 (file:///home/travis/build/overvenus/grpc-rs)
error: linking with `cc` failed: exit code: 1
  |
  = note: "cc" "-Wl,--as-needed" "-Wl,-z,noexecstack" "-m64" "-L" "/home/travis/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "/home/travis/build/overvenus/grpc-rs/target/debug/deps/grpcio_sys-6001af5e1c5efae1.1v9koybaboeosxiy.rcgu.o" "/home/travis/build/overvenus/grpc-rs/target/debug/deps/grpcio_sys-6001af5e1c5efae1.1y16o1qfye96o7m0.rcgu.o" "/home/travis/build/overvenus/grpc-rs/target/debug/deps/grpcio_sys-6001af5e1c5efae1.2rnarwe9869ff3em.rcgu.o" "/home/travis/build/overvenus/grpc-rs/target/debug/deps/grpcio_sys-6001af5e1c5efae1.2xs38k3nu4oe9uo6.rcgu.o" "/home/travis/build/overvenus/grpc-rs/target/debug/deps/grpcio_sys-6001af5e1c5efae1.3rngp6bm2u2q5z0y.rcgu.o" "/home/travis/build/overvenus/grpc-rs/target/debug/deps/grpcio_sys-6001af5e1c5efae1.4oc10dk278mpk1vy.rcgu.o" "/home/travis/build/overvenus/grpc-rs/target/debug/deps/grpcio_sys-6001af5e1c5efae1.4xq48u46a1pwiqn7.rcgu.o" "/home/travis/build/overvenus/grpc-rs/target/debug/deps/grpcio_sys-6001af5e1c5efae1.51s1w397y42gpez1.rcgu.o" "/home/travis/build/overvenus/grpc-rs/target/debug/deps/grpcio_sys-6001af5e1c5efae1.5et8vp13oj1m4v7a.rcgu.o" "/home/travis/build/overvenus/grpc-rs/target/debug/deps/grpcio_sys-6001af5e1c5efae1.8xzrsc1ux72v29j.rcgu.o" "/home/travis/build/overvenus/grpc-rs/target/debug/deps/grpcio_sys-6001af5e1c5efae1.9elsx31vb4it187.rcgu.o" "/home/travis/build/overvenus/grpc-rs/target/debug/deps/grpcio_sys-6001af5e1c5efae1.oa3rad818d8sgn4.rcgu.o" "-o" "/home/travis/build/overvenus/grpc-rs/target/debug/deps/grpcio_sys-6001af5e1c5efae1" "/home/travis/build/overvenus/grpc-rs/target/debug/deps/grpcio_sys-6001af5e1c5efae1.crate.allocator.rcgu.o" "-Wl,--gc-sections" "-pie" "-Wl,-z,relro,-z,now" "-nodefaultlibs" "-L" "/home/travis/build/overvenus/grpc-rs/target/debug/deps" "-L" "/home/travis/.cache/lib" "-L" "/home/travis/.cache/lib" "-L" "/home/travis/build/overvenus/grpc-rs/target/debug/build/grpcio-sys-afd36be57b22e64e/out" "-L" "/home/travis/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-l" "grpc" "-l" "grpc++" "-Wl,-Bstatic" "-Wl,--whole-archive" "-l" "grpc_wrap" "-Wl,--no-whole-archive" "-Wl,-Bdynamic" "-l" "stdc++" "-Wl,-Bstatic" "/home/travis/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libtest-315d48fd46f254fe.rlib" "/home/travis/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libterm-4dd46751eaaa3cb4.rlib" "/home/travis/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libgetopts-b0cb061f9f47a202.rlib" "/home/travis/build/overvenus/grpc-rs/target/debug/deps/liblibc-76755a6888618a72.rlib" "-Wl,--start-group" "/home/travis/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-2e0e6fd5a64772ec.rlib" "/home/travis/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libpanic_unwind-731e169e757c346a.rlib" "/home/travis/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liballoc_jemalloc-095fc2295ae9ef1b.rlib" "/home/travis/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libunwind-937988d834c61738.rlib" "/home/travis/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liballoc_system-b7ac4cdefbcbe74e.rlib" "/home/travis/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liblibc-2acd14cf338f9037.rlib" "/home/travis/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liballoc-abc00040df319771.rlib" "/home/travis/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcore-c03be6943b237af4.rlib" "-Wl,--end-group" "/home/travis/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler_builtins-7d0cd09e2c099c63.rlib" "-Wl,-Bdynamic" "-l" "util" "-l" "util" "-l" "dl" "-l" "rt" "-l" "pthread" "-l" "pthread" "-l" "gcc_s" "-l" "c" "-l" "m" "-l" "rt" "-l" "pthread" "-l" "util" "-l" "util"
  = note: /home/travis/build/overvenus/grpc-rs/target/debug/build/grpcio-sys-afd36be57b22e64e/out/libgrpc_wrap.a(grpc_wrap.o):(.data.rel._ZN4grpc8internalL14g_core_codegenE+0x0): undefined reference to `vtable for grpc::CoreCodegen'
          /home/travis/build/overvenus/grpc-rs/target/debug/build/grpcio-sys-afd36be57b22e64e/out/libgrpc_wrap.a(grpc_wrap.o): In function `grpc::CoreCodegen::~CoreCodegen()':
          /home/travis/.cache/include/grpcpp/impl/codegen/core_codegen.h:32: undefined reference to `vtable for grpc::CoreCodegen'
          collect2: error: ld returned 1 exit status
```

It may be related to https://github.com/grpc/grpc/pull/13500

Reply:
Yes I see the same error:

```
   Compiling clap v2.31.2
   Compiling grpcio-sys v0.2.3 (file:///home/hoverbear/git/grpc-rs/grpc-sys)
   Compiling grpcio-compiler v0.3.0 (file:///home/hoverbear/git/grpc-rs/compiler)
   Compiling grpcio v0.3.0 (file:///home/hoverbear/git/grpc-rs)
   Compiling grpcio-proto v0.3.0 (file:///home/hoverbear/git/grpc-rs/proto)
error: linking with `cc` failed: exit code: 1
  |
  = note: "cc" "-Wl,--as-needed" "-Wl,-z,noexecstack" "-m64" "-L" "/home/hoverbear/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "/home/hoverbear/git/grpc-rs/target/debug/deps/grpcio_sys-04563ee405ba0b03.1v9koybaboeosxiy.rcgu.o" "/home/hoverbear/git/grpc-rs/target/debug/deps/grpcio_sys-04563ee405ba0b03.1y16o1qfye96o7m0.rcgu.o" "/home/hoverbear/git/grpc-rs/target/debug/deps/grpcio_sys-04563ee405ba0b03.2rnarwe9869ff3em.rcgu.o" "/home/hoverbear/git/grpc-rs/target/debug/deps/grpcio_sys-04563ee405ba0b03.2xs38k3nu4oe9uo6.rcgu.o" "/home/hoverbear/git/grpc-rs/target/debug/deps/grpcio_sys-04563ee405ba0b03.3rngp6bm2u2q5z0y.rcgu.o" "/home/hoverbear/git/grpc-rs/target/debug/deps/grpcio_sys-04563ee405ba0b03.4oc10dk278mpk1vy.rcgu.o" "/home/hoverbear/git/grpc-rs/target/debug/deps/grpcio_sys-04563ee405ba0b03.4xq48u46a1pwiqn7.rcgu.o" "/home/hoverbear/git/grpc-rs/target/debug/deps/grpcio_sys-04563ee405ba0b03.51s1w397y42gpez1.rcgu.o" "/home/hoverbear/git/grpc-rs/target/debug/deps/grpcio_sys-04563ee405ba0b03.5et8vp13oj1m4v7a.rcgu.o" "/home/hoverbear/git/grpc-rs/target/debug/deps/grpcio_sys-04563ee405ba0b03.8xzrsc1ux72v29j.rcgu.o" "/home/hoverbear/git/grpc-rs/target/debug/deps/grpcio_sys-04563ee405ba0b03.9elsx31vb4it187.rcgu.o" "/home/hoverbear/git/grpc-rs/target/debug/deps/grpcio_sys-04563ee405ba0b03.oa3rad818d8sgn4.rcgu.o" "-o" "/home/hoverbear/git/grpc-rs/target/debug/deps/grpcio_sys-04563ee405ba0b03" "/home/hoverbear/git/grpc-rs/target/debug/deps/grpcio_sys-04563ee405ba0b03.crate.allocator.rcgu.o" "-Wl,--gc-sections" "-pie" "-Wl,-z,relro,-z,now" "-nodefaultlibs" "-L" "/home/hoverbear/git/grpc-rs/target/debug/deps" "-L" "/usr/lib" "-L" "/usr/lib" "-L" "/home/hoverbear/git/grpc-rs/target/debug/build/grpcio-sys-ac23585aa46eac93/out" "-L" "/home/hoverbear/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-l" "grpc" "-l" "grpc++" "-Wl,-Bstatic" "-Wl,--whole-archive" "-l" "grpc_wrap" "-Wl,--no-whole-archive" "-Wl,-Bdynamic" "-l" "stdc++" "-Wl,-Bstatic" "/home/hoverbear/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libtest-019326da6686230a.rlib" "/home/hoverbear/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libterm-7f8a78dfff7bac57.rlib" "/home/hoverbear/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libgetopts-dbfbcb47517df890.rlib" "/home/hoverbear/git/grpc-rs/target/debug/deps/liblibc-77d5831ef1267f52.rlib" "-Wl,--start-group" "/home/hoverbear/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-0888da25bc5aa659.rlib" "/home/hoverbear/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libpanic_unwind-c21a3f7e9b2ce71d.rlib" "/home/hoverbear/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liballoc_jemalloc-a89ec74248f6f925.rlib" "/home/hoverbear/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libunwind-9232e45013d81e94.rlib" "/home/hoverbear/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liballoc_system-afbe0a28ca4ea99f.rlib" "/home/hoverbear/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liblibc-bcf95ff3aa4e3276.rlib" "/home/hoverbear/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liballoc-26716181a02171e3.rlib" "/home/hoverbear/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcore-978afee7e21437ec.rlib" "-Wl,--end-group" "/home/hoverbear/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler_builtins-3f4ad009a7cf735f.rlib" "-Wl,-Bdynamic" "-l" "util" "-l" "util" "-l" "dl" "-l" "rt" "-l" "pthread" "-l" "pthread" "-l" "gcc_s" "-l" "c" "-l" "m" "-l" "rt" "-l" "pthread" "-l" "util" "-l" "util"
  = note: /home/hoverbear/git/grpc-rs/target/debug/build/grpcio-sys-ac23585aa46eac93/out/libgrpc_wrap.a(grpc_wrap.o):(.data.rel._ZN4grpc8internalL14g_core_codegenE+0x0): undefined reference to `vtable for grpc::CoreCodegen'
          /home/hoverbear/git/grpc-rs/target/debug/build/grpcio-sys-ac23585aa46eac93/out/libgrpc_wrap.a(grpc_wrap.o): In function `grpc::CoreCodegen::~CoreCodegen()':
          /usr/include/grpcpp/impl/codegen/core_codegen.h:32: undefined reference to `vtable for grpc::CoreCodegen'
          collect2: error: ld returned 1 exit status
```

Reply:
do we need to wait for the official team fixing it or just work around it? @overvenus

Reply:
It have been fixed in e2639c3.